src/state/registers.ts

```diff
@@ -583,6 +591,37 @@ export abstract class RegisterSet implements vscode.Disposable {
 
     this._named.set("", this.null);
     this._named.set("null", this.null);
+
+    // Watch systemClipboardRegister setting and update binding
+    if (extension !== undefined) {
+      extension.observePreference<keyof Registers>(
+        ".systemClipboardRegister",
+        (value) => {
+          if (!['"', null, ...availableClipboardRegisters].includes(value)) {
+            return;
```

Comment:
I think this should be equivalent to having no system clipboard register.

```suggestion
            value = null;
```

Reply:
And logging a warning.